src/v/cluster/partition.cc

```diff
@@ -1165,6 +1165,100 @@ ss::future<> partition::unsafe_reset_remote_partition_manifest(iobuf buf) {
       ntp());
 }
 
+ss::future<> partition::unsafe_reset_remote_partition_manifest_from_cloud() {
```

Comment:
It's a strange place to place cloud download logic. I think that this can be a method in the `ntp_archiver`.

Reply:
The method can pause the workflow when it starts.

Reply:
Hm. I've reworked this to stop the archiver and then perform the download and reset. I can really do it in the archiver since, I'm stopping and destroying it before the download.

Reply:
that makes sense short term